sentinel/pkg/checker/event/app.go

```diff
@@ -121,20 +121,36 @@ func checkFeeds(ctx context.Context, FeedsToCheck []FeedToCheck) {
 	msg := ""
 	for i, feed := range FeedsToCheck {
 		offset, err := timeSinceLastFeedEvent(ctx, feed)
-		if err != nil {
+		if err == nil {
+			if offset > time.Duration(feed.ExpectedInterval)*time.Millisecond*2 {
+				log.Warn().Str("feed", feed.FeedName).Msg(fmt.Sprintf("%s delayed by %s", feed.FeedName, offset-time.Duration(feed.ExpectedInterval)*time.Millisecond))
+				FeedsToCheck[i].LatencyChecked++
+				if FeedsToCheck[i].LatencyChecked > AlarmOffset {
+					msg += fmt.Sprintf("%s delayed by %s\n", feed.FeedName, offset-time.Duration(feed.ExpectedInterval)*time.Millisecond)
+					FeedsToCheck[i].LatencyChecked = 0
+				}
+			} else {
+				FeedsToCheck[i].LatencyChecked = 0
+			}
+		} else {
 			log.Error().Err(err).Str("feed", feed.FeedName).Msg("Failed to check feed")
-			continue
 		}
 
-		if offset > time.Duration(feed.ExpectedInterval)*time.Millisecond*2 {
-			log.Warn().Str("feed", feed.FeedName).Msg(fmt.Sprintf("%s delayed by %s", feed.FeedName, offset-time.Duration(feed.ExpectedInterval)*time.Millisecond))
-			FeedsToCheck[i].LatencyChecked++
-			if FeedsToCheck[i].LatencyChecked > AlarmOffset {
-				msg += fmt.Sprintf("%s delayed by %s\n", feed.FeedName, offset-time.Duration(feed.ExpectedInterval)*time.Millisecond)
-				FeedsToCheck[i].LatencyChecked = 0
+		count, err := countLastMinFeedEvents(ctx, feed)
+		if err == nil {
+			maxFeedSubmissionCount := int(time.Minute.Milliseconds() / (time.Duration(feed.ExpectedInterval) * time.Millisecond).Milliseconds()) * 2
```

Comment:
```suggestion
			maxFeedSubmissionCount := int(time.Minute.Milliseconds() / (time.Duration(feed.ExpectedInterval) * time.Millisecond)) * 2
```

I think it'd work without `Milliseconds()` conversion at the last part

Reply:
when I remove it, i get the following error:
```
invalid operation: time.Minute.Milliseconds() / (time.Duration(feed.ExpectedInterval) * time.Millisecond) (mismatched types int64 and time.Duration)compiler[MismatchedTypes](https://pkg.go.dev/golang.org/x/tools/internal/typesinternal#MismatchedTypes)
```

Reply:
interesting 🤔 let's keep it as it is